plugins/available/z_autoenv.plugin.bash

```diff
@@ -9,35 +9,36 @@ else
 	__array_offset=1
 fi
 
-function autoenv_init {
-	typeset home _file
+autoenv_init() {
+	typeset home _file # target
 	typeset -a _files
+	#target=$1
 	home="${HOME%/*}"
 
-	_files=($(
-		while [[ "$PWD" != "/" && "$PWD" != "$home" ]]; do
-			_file="$PWD/.env"
-			if [[ -e "${_file}" ]]; then
-				echo "${_file}"
-			fi
-			builtin cd .. || return
-		done
-	))
+	while [[ "$PWD" != "/" && "$PWD" != "$home" ]]; do
+		_file="$PWD/.env"
+		if [[ -e "${_file}" ]]; then
+			_files+=("${_file}")
+		fi
+		builtin cd .. || true
+	done
 
 	_file=${#_files[@]}
 	while ((_file > 0)); do
-		# shellcheck disable=SC1090
+		#shellcheck disable=SC1090
 		source "${_files[_file - __array_offset]}"
 		: $((_file -= 1))
 	done
 }
 
 cd() {
+	local return_code
 	if builtin cd "$@"; then
 		autoenv_init
 		return 0
 	else
+		return_code=$?
 		echo "else?"
-		return $?
+		return $return_code
```

Comment:
I'm unsure about the coding guideline of this project, but I generally suggest to quote it as
```suggestion
		return "$return_code"
```

in case `IFS=0123456789`.

Reply:
the pre-commit doesn't complain but I don't mind adding it. you and I know it's just a number and bash is not exactly strongly typed.